pkg/repository/sql/config.go

```diff
@@ -0,0 +1,40 @@
+package sql
+
+import (
+	"fmt"
+
+	"github.com/ThreeDotsLabs/esja/pkg/aggregate"
+)
+
+type Config[T any] struct {
```

Comment:
👍 

Nice!

I understand the Serializer will get split at some point in the near future.